src/tests/ftest/control/dmg_pool_query_test.py

```diff
@@ -0,0 +1,177 @@
+            "scm_info": self.params.get("scm_info", "/run/exp_vals/"),
+            "nvme_info": self.params.get("nvme_info", "/run/exp_vals/"),
+            "rebuild_info": self.params.get("rebuild_info", "/run/exp_vals/"),
+        }
+
+        # Add the expected uuid value
+        exp_info["pool_info"].insert(0, self.uuid.upper())
+
+        if exp_info != dmg_info:
+            self.log.info("==>   Found difference in dmg output and expected.")
+            self.fail("dmg pool-query: \n{} \nexpected: \n{}".format(
+                dmg_info, exp_info))
+        else:
+            self.log.info("==>   Expect values found in dmg pool query output.")
+
+    def test_pool_query_inputs(self):
+        """
+        JIRA ID: DAOS-2976
+
+        Test Description: Test basic dmg functionality to query pool info on
+        the system. Verify the inputs that can be provided to 'query --pool'
+        argument of the dmg pool subcommand.
+
+        :avocado: tags=all,small,pr,hw,dmg,pool_query,basic,poolqueryinputs
+        """
+        # Get test UUIDs
+        errors_list = []
+        uuids = self.params.get("uuids", '/run/pool_uuids/*')
+
+        # Disable raising an exception if the dmg command fails
+        self.dmg.exit_status_exception = False
+
+        for uuid in uuids:
+            self.log.info("\n==>   Using test UUID: %s", uuid[0])
+            self.log.info("==>   Test is expected to finish with: %s", uuid[1])
+
+            # Verify
+            if self.get_pool_query_info(uuid[0]):
+                exception = None
+            elif not self.get_pool_query_info(uuid[0]):
+                exception = 1
+
+            if uuid[1] == "FAIL" and exception is None:
+                errors_list.append("==>   Test expected to fail:" + uuid[0])
+            elif uuid[1] == "PASS" and exception:
+                errors_list.append("==>   Test expected to pass: " + uuid[0])
+
+        # Enable exceptions again for dmg.
+        self.dmg.exit_status_exception = True
+
+        # Report errors and fail test if needed.
+        if errors_list:
+            for err in errors_list:
+                self.log.error("==>   Failure: %s", err)
+            self.fail("Failed dmg pool query input test.")
+
+    def test_pool_query_ior(self):
+        """
+        JIRA ID: DAOS-2976
+
+        Test Description: Test that pool query command will properly and
+        accurately show the size changes once there is content in the pool.
+
+        :avocado: tags=all,small,pr,hw,dmg,pool_query,basic,poolquerywrite
+        """
+        # Store orignal pool info
+        out_b = self.get_pool_query_info(self.uuid)
+        self.log.info("==>   Pool info before write: \n%s", out_b)
+
+        #  Run ior
+        self.log.info("==>   Write data to pool.")
+        self.run_ior_with_pool()
+
+        # Check pool written data
+        out_a = self.get_pool_query_infself.get_pool_query_info(self.uuid)
```

Comment:
(pylint-no-member) Instance of 'DmgPoolQueryTest' has no 'get_pool_query_infself' member